lib/clAdapters/google/errors.ts

```diff
@@ -0,0 +1,40 @@
+export abstract class GoogleError extends Error {
```

Comment:
@bsouthga: I don't disagree with the discriminated unions in general, but is it possible to do the discrimination if at runtime we might get regular `Error`s from 3rd party code? I'm not sure we can use an `instanceof` or `typeof` check in this case. E.g.
```ts
function handleGoogleError(googErr: GoogleError | Error) { // In case random code throws an error
  // if (googErr instanceof GoogleError) <-- GoogleError is just a type
  // if (typeof googErr === 'GoogleError')
    switch (googErr.kind) {
        case GoogleErrorType.InvalidGrant: {
        ...
  // else
  //    Process as regular Error
```
I think to get around above, all adapter code would have to catch all errors and recreate as a `GoogleError` - is that what you were thinking? Also then in our application code, would we need a super union type? Something like the following?
```ts
type AdapterError
  = GoogleError | Office365Error | JiraError;
```